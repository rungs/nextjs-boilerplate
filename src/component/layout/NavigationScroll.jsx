import { useEffect } from 'react'
import { useRouter } from 'next/router'

const NavigationScroll = ({ children }) => {
  const location = useRouter()
  const { pathname } = location

  useEffect(() => {
    window.scrollTo({
      top: 0,
      left: 0,
      behavior: 'smooth',
    })
  }, [pathname])

  return children || null
}

export default NavigationScroll
