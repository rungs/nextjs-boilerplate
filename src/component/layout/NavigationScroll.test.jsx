import React from 'react'
import { render } from '@testing-library/react'
import NavigationScroll from './NavigationScroll'

jest.mock('next/router', () => ({
  useRouter: () => ({
    pathname: '/mocked-path', // Replace with the desired pathname for your test
  }),
}))

describe('Test Component NavigationScroll', () => {
  afterEach(() => {
    jest.clearAllMocks()
  })

  it('NavigationScroll component snapshot', () => {
    const { asFragment } = render(<NavigationScroll />)
    expect(asFragment()).toMatchSnapshot()
  })
})
