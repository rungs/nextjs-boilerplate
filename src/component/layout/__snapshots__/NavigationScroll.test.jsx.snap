// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Component NavigationScroll NavigationScroll component snapshot 1`] = `<DocumentFragment />`;
